tests/unit/test_param_escaper.py

```diff
@@ -0,0 +1,108 @@
+from datetime import date, datetime
+import unittest, pytest
+
+from databricks.sql.utils import ParamEscaper, inject_parameters
+
+pe = ParamEscaper()
+
+class TestIndividualFormatters(object):
+
+    # Test individual type escapers
+    def test_escape_number_integer(self):
+        """This behaviour falls back to Python's default string formatting of numbers
+        """
+        assert pe.escape_number(100) == 100
+
+    def test_escape_number_float(self):
+        """This behaviour falls back to Python's default string formatting of numbers
+        """
+        assert pe.escape_number(100.1234) == 100.1234
+
+    def test_escape_string_normal(self):
+        """
+        """
+
+        assert pe.escape_string("golly bob howdy") == "'golly bob howdy'"
+
+    def test_escape_string_that_includes_quotes(self):
+        # Databricks queries support just one special character: a single quote mark
+        # These are escaped by doubling: 
+        # e.g. INPUT: his name was 'robert palmer'
+        # e.g. OUTPUT: 'his name was ''robert palmer'''
+
+        assert pe.escape_string("his name was 'robert palmer'") == "'his name was ''robert palmer'''"
+
+    def test_escape_date_time(self):
+        INPUT = datetime(1991,8,3,21,55)
+        OUTPUT = "1991-08-03 21:55:00"
```

Comment:
Let's push this work to a later PR.  Right now, the parameter escaper doesn't handle timezones (it silently drops them). Since the goal of this PR is to document the existing functionality, the existing test is sufficient. And after the Databricks thrift server supports server-side parameter binding it won't be relevant. 

If users run into issues around this we can consider improving the binding behaviour. But it's low priority for now.